### PR TITLE
Rename md5 and sha1 to weak variants

### PR DESCRIPTION
**Pull request by asaintpi**
Updates made:

Updated name from sha1 to weak-sha1 in hash.rs and hmac.rs
Updated name from md5 to weak-md5 in hash.rs and hmac.rs
Updated documentation around weak-sha1 and weak-md5
Updated Cargo.toml to reflect.

### CODE REVIEW

File: rust-symcrypt/Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ lazy_static = "1.4.0"
 default = []
 md5 = []
 sha1 = []
```
Comment (nnmkhang) [inline rust-symcrypt/Cargo.toml:24]: these should be `weak-sha1` and `weak-md5`. You were correct with changing the `weak-crypto` feature, but right now when you apply `weak-crypto` you are enabling `md5` and `sha1` which is not implemented anywhere in the code

File: rust-symcrypt/src/hash.rs [hunk 2]
```diff
@@ -535,7 +535,7 @@ pub fn sha512(data: &[u8]) -> [u8; SHA512_RESULT_SIZE] {
 
 #[cfg(test)]
 mod test {
-    // Note: by default sha1 and md5 are turned off, to enable for testing you can use:
+    // Note: by default "weak-sha1" and "weak-md5" are turned off, to enable for testing you can use:
     // cargo test --features "weak-crypto"
```
Comment (nnmkhang) [inline rust-symcrypt/src/hash.rs:539]: did you try running this command and verifying that the tests are being called after your changes?

**Comment by nnmkhang** [comment]
Please update the documentation on `README.md` and `lib.rs` to communicate the new naming convention

**Review by nnmkhang — CHANGES_REQUESTED** [review]
LGTM,

Left a few comments. 

Double check the Cargo.toml as well as run the test with the flag enabled to see that the flags are being called properly 

also nitpick: You should name your branches in a way that is tied to the user more clearly. ie: user/asauntpi/weak-md5-sha1-names

**Review by mmwuyan — APPROVED** [review]
aside from Khang's comments, lgtm

**Pull request closed without merging**